src/sentry/api/endpoints/internal/integration_proxy.py

```diff
@@ -102,9 +105,9 @@ def _validate_request(self, request: HttpRequest) -> bool:
             organization_id=self.org_integration.organization_id
         )
         self.client: IntegrationProxyClient = installation.get_client()
-        client_type = type(self.client)
-        self.log_extra["client_type"] = client_type.__name__
-        if not issubclass(client_type, IntegrationProxyClient):
+        client_class = self.client.__class__
+        self.log_extra["client_type"] = client_class.__name__
+        if not issubclass(client_class, IntegrationProxyClient):
```

Comment:
@leeandher I needed to do this to properly mock the client in a test here https://github.com/getsentry/sentry/blob/72a529131b03f8689ff4f9b7b67924ce6a5418eb/tests/sentry/api/endpoints/internal/test_integration_proxy.py#L92

Changing this to `self.client.__class__` made this easier to mock:

> If spec is an object (rather than a list of strings) then [`__class__`](https://docs.python.org/3/library/stdtypes.html#instance.__class__) returns the class of the spec object. This allows mocks to pass [isinstance()](https://docs.python.org/3/library/functions.html#isinstance) tests.